bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/util/RuntimeUtils.java

```diff
@@ -79,7 +73,7 @@ public static void addToArray(Type type, String value, ArrayValue array) {
         // TODO: need to add parsing logic for ref values for both var args and other args as well.
         switch (type.getTag()) {
             case TypeTags.STRING_TAG:
-                array.add(array.size(), value);
+                array.add(array.size(), StringUtils.fromString(value));
```

Comment:
Shall we write some tests to cover those missing lines?

Reply:
This is a refactoring I did. Not related to the main code